packages/trace-viewer/src/ui/snapshotTab.tsx

```diff
@@ -227,37 +227,54 @@ export const InspectModeController: React.FunctionComponent<{
   iteration: number,
 }> = ({ iframe, isInspecting, sdkLanguage, testIdAttributeName, highlightedLocator, setHighlightedLocator, iteration }) => {
   React.useEffect(() => {
-    const win = iframe?.contentWindow as any;
-    let recorder: Recorder | undefined;
+    const recorders: { recorder: Recorder, frameSelector: string }[] = [];
     try {
-      if (!win)
-        return;
-      recorder = win._recorder;
-      if (!recorder && !isInspecting && !highlightedLocator)
-        return;
+      createRecorders(recorders, sdkLanguage, testIdAttributeName, '', iframe?.contentWindow);
     } catch {
-      // Potential cross-origin exception when accessing win._recorder.
-      return;
+      // Potential cross-origin exceptions.
     }
-    if (!recorder) {
-      const injectedScript = new InjectedScript(win, false, sdkLanguage, testIdAttributeName, 1, 'chromium', []);
-      recorder = new Recorder(injectedScript, {
+
+    for (const { recorder, frameSelector } of recorders) {
+      const actionSelector = locatorOrSelectorAsSelector(sdkLanguage, highlightedLocator, testIdAttributeName);
+      recorder.setUIState({
+        mode: isInspecting ? 'inspecting' : 'none',
+        actionSelector,
```

Comment:
I wonder what happens when `actionSelector` has a `.frameLocator()` inside? For example, what if user edits iframe's locator inside the codemirror?